python/src/filesystem.rs

```diff
@@ -41,21 +41,27 @@ impl DeltaFileSystemHandler {
 #[pymethods]
 impl DeltaFileSystemHandler {
     #[new]
-    #[pyo3(signature = (table_uri, options = None, known_sizes = None))]
+    #[pyo3(signature = (table_uri, table = None, options = None, known_sizes = None))]
     fn new(
-        table_uri: &str,
+        table_uri: String,
+        table: Option<&RawDeltaTable>,
```

Comment:
Maybe instead make a separate classmethod `from_table` that takes the table? Then perhaps we can keep the pickling support.